File: apps/backend/app/schemas/product_category.py
from uuid import UUID
from datetime import datetime
from typing import List
from pydantic import BaseModel, ConfigDict

class ProductCategoryBase(BaseModel):
    name: str
    slug: str

class ProductCategoryCreate(ProductCategoryBase):
    pass

class ProductCategoryRead(ProductCategoryBase):
    id: UUID
    company_id: UUID
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class PaginatedProductCategories(BaseModel):
    total: int
    skip: int
    limit: int
    items: List[ProductCategoryRead]

    model_config = ConfigDict()

File: apps/portal-company/src/types/productCategory.ts
// src/types/productCategory.ts

export interface ProductCategoryBase {
  name: string;
  slug: string;
}

// Create/Update são aliases do Base, não interfaces vazias
export type ProductCategoryCreate = ProductCategoryBase;
export type ProductCategoryUpdate = ProductCategoryBase;

/** Entidade retornada pela API */
export interface ProductCategoryRead extends ProductCategoryBase {
  id: string;          // UUID
  company_id: string;  // UUID da empresa
  created_at: string;  // ISO datetime
}

/** Resposta paginada */
export interface PaginatedProductCategories {
  total: number;
  skip:  number;
  limit: number;
  items: ProductCategoryRead[];
}

export interface ProductCategoryBasic {
  id: string;
  name: string;
}